devel/mimalloc/files/0002-Easy-on-tests-for-PPC.patch

```diff
@@ -0,0 +1,47 @@
+From e636851eeb2f51bdd2eab7cfd8f4a4d34cb4f16d Mon Sep 17 00:00:00 2001
+From: barracuda156 <vital.had@gmail.com>
+Date: Fri, 7 Jul 2023 21:19:13 +0800
+Subject: [PATCH 2/2] Easy on tests for PPC
+
+---
+ test/test-api.c    | 10 +++++-----
+ test/test-stress.c |  2 +-
+ 2 files changed, 6 insertions(+), 6 deletions(-)
+
+diff --git test/test-api.c test/test-api.c
+index 8dd24e1..5d4c763 100644
+--- test/test-api.c
++++ test/test-api.c
+@@ -295,11 +295,11 @@ int main(void) {
+   // ---------------------------------------------------
+   // various
+   // ---------------------------------------------------
+-  CHECK_BODY("realpath") {
+-    char* s = mi_realpath( ".", NULL );
+-    // printf("realpath: %s\n",s);
+-    mi_free(s);
+-  };
++//   CHECK_BODY("realpath") {
++//     char* s = mi_realpath( ".", NULL );
++//     // printf("realpath: %s\n",s);
++//     mi_free(s);
++//   };
```

Comment:
Why is this commented?

Reply:
This test case fails on PPC. (Again, should not be applied to other systems, sorry, I was too tired yesterday.)

Reply:
No worries. I'm curious why it fails though, it seems like a simple test.

Reply:
BTW it passed fine when `MI_OSX_ZONE` is on, but then stress test is broken.

Reply:
I added comments to the patch re changes done.

Reply:
This is strange tbh. This function is part of the public API so it should work. See if you can investigate further - which line in the source does it break on?

Reply:
The source for the function is [here](https://github.com/microsoft/mimalloc/blob/v2.1.2/src/alloc.c#L864). It doesn't do anything particularly crazy.

Reply:
Let me try with the debugger.